src/ui/components/Library/TeamButton.tsx

```diff
@@ -7,18 +7,21 @@ import hooks from "ui/hooks";
 import SidebarButton from "./SidebarButton";
 import classNames from "classnames";
 import { Redacted } from "../Redacted";
+import { Workspace } from "ui/types";
 
 type TeamButtonProps = PropsFromRedux & {
   text: string;
   isNew?: boolean;
   id: string | null;
+  workspace?: Workspace;
```

Comment:
Refactoring idea: There's a bit of mixed responsibilities here where the consumer passes `isNew` to trigger the badge or passes the `workspace` and this comp triggers the badge. Instead, the consumer could pass a badge prop (`"new" | "trial" | undefined`) and the component could simplified.

I might take a cut at this separately; not a blocking concern